hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/NodeEndpoint.java

```diff
@@ -171,4 +181,163 @@ private DatanodeStorageReport getStorageReport(DatanodeDetails datanode) {
+          } else {
+            Response.ResponseBuilder builder = Response.status(Response.Status.BAD_REQUEST);
+            builder.entity("{\n" +
+                "    \"Invalid request: Pre-checks failed for selected datanodes. DataNode should pass following " +
+                "pre-checks.\": [\n" +
+                "        {\n" +
+                "            \"title\": \"Incorrect State\",\n" +
+                "            \"description\": \"DataNode should be in either DECOMMISSIONED operational state or " +
+                "DEAD node state.\"\n" +
+                "        },\n" +
+                "        {\n" +
+                "            \"title\": \"Open Containers\",\n" +
+                "            \"description\": \"Containers are open for few or all selected datanodes.\"\n" +
+                "        },\n" +
+                "        {\n" +
+                "            \"title\": \"Open Pipelines\",\n" +
+                "            \"description\": \"Pipelines are open for few or all selected datanodes.\"\n" +
+                "        }\n" +
+                "    ]\n" +
+                "}");
+            failedDatanodes.add(DatanodeMetadata.newBuilder()
+                .withHostname(nodeManager.getHostName(nodeByUuid))
+                .withUUid(uuid)
+                .withOperationalState(nodeByUuid.getPersistedOpState())
+                .withState(nodeManager.getNodeStatus(nodeByUuid).getHealth())
+                .build());
+            // Build and return the response
+            return builder.build();
+          }
+        } catch (NodeNotFoundException nnfe) {
+          LOG.error("Selected node {} not found : {} ", uuid, nnfe);
+          notFoundDatanodes.add(DatanodeMetadata.newBuilder()
+                  .withHostname("")
+                  .withState(NodeState.DEAD)
+              .withUUid(uuid).build());
+        }
+      }
+    } catch (Exception exp) {
+      LOG.error("Unexpected Error while removing datanodes : {} ", exp);
+      throw new WebApplicationException(exp, Response.Status.INTERNAL_SERVER_ERROR);
+    }
+
+    RemoveDataNodesResponseWrapper removeDataNodesResponseWrapper = new RemoveDataNodesResponseWrapper();
+
+    if (!failedDatanodes.isEmpty()) {
+      DatanodesResponse failedNodesResp =
+          new DatanodesResponse(failedDatanodes.size(), failedDatanodes);
+      failedNodesResp.setMessage("Invalid request: Nodes should be in either DECOMMISSIONED or " +
+          "IN_MAINTENANCE mode or in DEAD State.");
+      removeDataNodesResponseWrapper.getDatanodesResponseMap().put("failedDatanodes", failedNodesResp);
+    }
+
+    if (!notFoundDatanodes.isEmpty()) {
+      DatanodesResponse notFoundNodesResp =
+          new DatanodesResponse(notFoundDatanodes.size(), notFoundDatanodes);
+      notFoundNodesResp.setMessage("Invalid request: Selected nodes not found. Kindly send correct node " +
+          "details to remove it !!!");
```

Comment:
we can remove message, as it provide redundant information, as notFoundNodeResp object itself tell same if list is not empty

Reply:
Done.